hotel_tourist_tax/models/hotel_folio.py

```diff
@@ -0,0 +1,75 @@
+# See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class HotelFolio(models.Model):
+    _inherit = "hotel.folio"
+
+    @api.multi
+    def write(self, vals):
+        res = super(HotelFolio, self).write(vals)
+        for rec in self:
+            if (
+                bool(vals.get("duration"))
+                or bool(vals.get("adults"))
+                or bool(vals.get("children"))
+            ) and not bool(
+                vals.get("service_lines")  # prevent recursion
+                or rec.invoice_status == "invoiced"
+                or bool(rec.hotel_invoice_id)
+            ):
+                rec.update_tourist_tax()
```

Comment:
why not this formulation? (`in` keyword and useless boolean casting)

```py
if (
    "duration" in vals or "adults" in vals or "children" in vals
) and not (
    "service_lines" in vals 
    or rec.invoice_status == "invoiced" 
    or rec.hotel_invoice_id
):
```

Your version of the condition does not pass if `adults` and `children` are in `vals` and equals 0. But that might be what you want.

Reply:
Aha, much cleaner. This statement catcher more cases indeed, but that's a good thing. Thanks!